golang/rabin2.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+    "crypto/rand"
+    "fmt"
+    "hash/sha256"
+    "math/big"
+)
+
+// RabinPubKey is the public key for the Rabin signature algorithm.
+type RabinPubKey struct {
+    N *big.Int
+}
+
+// RabinSig is the signature for the Rabin signature algorithm.
+type RabinSig struct {
+    S *big.Int
+    U []byte
+}
+
+// GenerateRabinKeyPair generates a public and private key pair for the Rabin signature algorithm.
+func GenerateRabinKeyPair(securityLevel int) (*RabinPubKey, *big.Int, error) {
+    // Generate two large prime numbers.
+    p, err := rand.Prime(rand.Reader, securityLevel)
+    if err != nil {
+        return nil, nil, err
+    }
+
+    q, err := rand.Prime(rand.Reader, securityLevel)
+    if err != nil {
+        return nil, nil, err
+    }
+
+    // Compute the public key.
+    n := new(big.Int).Mul(p, q)
+
+    // Compute the private key.
+    phi := new(big.Int).Mul(p.Sub(p, big.NewInt(1)), q.Sub(q, big.NewInt(1)))
+    e := new(big.Int).SetUint64(65537)
+    d := new(big.Int).ModInverse(e, phi)
+
+    return &RabinPubKey{n}, d, nil
+}
+
+// SignRabin signs a message using the Rabin signature algorithm.
+func SignRabin(msg []byte, privKey *big.Int, pubKey *RabinPubKey) (*RabinSig, error) {
+    // Compute the hash of the message.
+    h := sha256.Sum256(msg)
```

Comment:
better to implement the hash function the same as TS, expand the hash result to N bits according to `securityLevel`

refer to https://github.com/sCrypt-Inc/rabin/blob/master/src/rabin.ts#L75-L89